spec/initializers/015_promote_upcoming_changes_spec.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+RSpec.describe "Promote upcoming changes initializer" do
+  context "when enable_upcoming_changes is disabled" do
+    before do
+      SiteSetting.enable_upcoming_changes = false
+      SiteSetting.promote_upcoming_changes_on_status = :stable
+    end
+
+    it "does nothing" do
+      SiteSetting.expects(:upcoming_change_site_settings).never
+      UpcomingChanges::AutoPromotionInitializer.call
+    end
+  end
+
+  context "when enable_upcoming_changes is enabled" do
+    before do
+      SiteSetting.enable_upcoming_changes = true
+      SiteSetting.upcoming_change_verbose_logging = true
+
+      mock_upcoming_change_metadata(
+        {
+          enable_upload_debug_mode: {
+            impact: "other,developers",
+            status: :stable,
+            impact_type: "other",
+            impact_role: "developers",
+          },
+          enable_user_tips: {
+            impact: "feature,all_members",
+            status: :beta,
+            impact_type: "feature",
+            impact_role: "all_members",
+          },
+          show_user_menu_avatars: {
+            impact: "feature,all_members",
+            status: :permanent,
+            impact_type: "feature",
+            impact_role: "all_members",
+          },
+        },
+      )
+    end
+
+    context "when the change does not meet the promotion criteria" do
+      before { SiteSetting.promote_upcoming_changes_on_status = :never }
+
+      it "does not enable the upcoming change and logs output" do
+        track_log_messages do |logger|
+          UpcomingChanges::AutoPromotionInitializer.call
+          expect(logger.warnings.join("\n")).to include(
+            "'enable_upload_debug_mode' did not meet promotion criteria. Current status is stable, required status is never.",
+          )
+          expect(logger.warnings.join("\n")).to include(
+            "'enable_user_tips' did not meet promotion criteria. Current status is beta, required status is never.",
+          )
+        end
+      end
+    end
+
+    context "when the upcoming change has already been set by the admin" do
+      before do
+        SiteSetting.enable_upload_debug_mode = false
+        SiteSetting.create!(
+          name: "enable_upload_debug_mode",
+          value: "f",
```

Comment:
You can even use `false` directly here, AR will convert it properly to `"f"` for the DB.